src/VisualStudio/CSharp/Impl/Options/Formatting/FormattingOptionPageControl.xaml.cs

```diff
@@ -29,6 +30,43 @@ public FormattingOptionPageControl(IServiceProvider serviceProvider) : base(serv
             BindToOption(FormatOnReturnCheckBox, FeatureOnOffOptions.AutoFormattingOnReturn, LanguageNames.CSharp);
             BindToOption(FormatOnPasteCheckBox, FeatureOnOffOptions.FormatOnPaste, LanguageNames.CSharp);
             SetNestedCheckboxesEnabled();
+
+            FormatDocumentSettingsGroupBox.Header = CSharpVSResources.Format_document_settings;
+            AllCSharpFormattingRulesCheckBox.Content = CSharpVSResources.All_csharp_formatting_rules;
+            RemoveUnusedUsingsCheckBox.Content = CSharpVSResources.Remove_unnecessary_usings;
+            SortUsingsCheckBox.Content = CSharpVSResources.Sort_usings;
+            FixImplicitExplicitTypeCheckBox.Content = CSharpVSResources.Fix_implicit_explicit_type;
+            FixThisQualificationCheckBox.Content = CSharpVSResources.Fix_this_qualification;
+            FixFrameworkTypesCheckBox.Content = CSharpVSResources.Fix_framework_types;
+            FixAddRemoveBracesCheckBox.Content = CSharpVSResources.Fix_add_remove_braces;
+            FixAccessibilityModifiersCheckBox.Content = CSharpVSResources.Fix_accessibility_modifiers;
+            SortAccessibilityModifiersCheckBox.Content = CSharpVSResources.Sort_accessibility_modifiers;
+            MakeReadonlyCheckBox.Content = CSharpVSResources.Make_readonly;
+            RemoveUnnecessaryCastsCheckBox.Content = CSharpVSResources.Remove_unnecessary_casts;
+            FixExpressionBodiedMembersCheckBox.Content = CSharpVSResources.Fix_expression_bodied_members;
+            FixInlineVariableDeclarationsCheckBox.Content = CSharpVSResources.Fix_inline_variable_declarations;
+            RemoveUnusedVariablesCheckBox.Content = CSharpVSResources.Remove_unused_variables;
+            FixObjectCollectionInitializationCheckBox.Content = CSharpVSResources.Fix_object_collection_initialization;
+            FixLanguageFeaturesCheckBox.Content = CSharpVSResources.Fix_language_features;
+
+            // IsCodeCleanupConfiguredCheckBox is hidden all the time, and it tracks if the user ever configured the code cleanup
+            BindToOption(AreCodeCleanupRulesConfiguredCheckBox, CodeCleanupOptions.AreCodeCleanupRulesConfigured, LanguageNames.CSharp);
```

Comment:
I think you don't need to do this but use the Option straight when needed.

Reply:
Cannot update the value of the Option directly, and that's why we have to use a hidden checkbox to do the magic